packages/adblocker/src/engine/engine.ts

```diff
@@ -229,6 +229,79 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
     return engine as InstanceType<T>;
   }
 
+  public static merge<T extends FilterEngine>(this: new (...args: any[]) => T, engines: T[]): T {
+    if (!engines || engines.length < 2) {
+      throw new Error('merging engines requires at least two engines');
+    }
+
+    const config = engines[0].config;
+    const lists = engines[0].lists;
+    const engineVersion = engines[0].engineVersion;
+
+    const networkFilters: Map<number, NetworkFilter> = new Map();
+    const cosmeticFilters: Map<number, CosmeticFilter> = new Map();
+    const preprocessors: Preprocessor[] = [];
+
+    const configKeysMustMatch: Exclude<keyof Config, 'serialize' | 'getSerializedSize'>[] = [
+      'debug',
+      'enableCompression',
+      'enableOptimizations',
+    ];
+
+    for (const engine of engines) {
+      // Validate the config
+      for (const configKey of configKeysMustMatch) {
+        if (config[configKey] !== engine.config[configKey]) {
+          throw new Error(`config "${configKey}" of all merged engines must be the same`);
+        }
+      }
+
+      if (engineVersion !== engine.engineVersion) {
+        throw new Error(
+          `engine version mismatch during merge, expected ${engineVersion} but got ${engine.engineVersion}`,
+        );
```

Comment:
After some thoughts I'm not sure if this is really required since the version is a global in the adblocker library and this would only be an issue if multiple versions of the library are loaded at the same time.

Reply:
This is now dropped in [`21e5350` (#4138)](https://github.com/ghostery/adblocker/pull/4138/commits/21e535016146ace54008216e4019ec8ba5bee10b)